### PR TITLE
PI-18622: RFS v2

### PR DESCRIPTION
**Pull request by AjayAujlaWork**
Recommended to hide whitespace changes since tabs were replaced by spaces: 
image

https://appdirect.jira.com/wiki/spaces/PI/pages/982647318/TD+Required+field+Service#3.4--Connector
https://appdirect.jira.com/wiki/spaces/PI/pages/1079148713/TD+Required+field+Service+-+Phase

https://github.com/AppDirect/service-integration-sdk/pull/216

#### Dependencies
[//]: # "optional, remove if none"
List additional Jira tickets or Github PRs

#### Description
Description of the PR. Make language appropriate for other engineers not involved.

#### Manual merge checklist:
- [ ] Code Review completed
- [ ] Code coverage
- [ ] QA Validation completed
  - Testrail TestCase/Plan link:
- [ ] Approved by a PI tech lead
- [ ] Github checks all green

### CODE REVIEW

File: src/main/java/com/appdirect/sdk/vendorFields/model/v2/FieldTypeV2.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package com.appdirect.sdk.vendorFields.model.v2;
+
+import lombok.Getter;
+
+@Getter
+public enum FieldTypeV2 {
+    DOMAIN, STRING, EMAIL, NAME, ZIP_CODE, PHONE_NUMBER, COUNTRY, STATE, TEXT
```
Comment (abidhkhan) [inline src/main/java/com/appdirect/sdk/vendorFields/model/v2/FieldTypeV2.java:7]: Why do we need it ?

Comment (abidhkhan) [reply src/main/java/com/appdirect/sdk/vendorFields/model/v2/FieldTypeV2.java:7]: got it,please ignore

Comment (AjayAujlaWork) [reply src/main/java/com/appdirect/sdk/vendorFields/model/v2/FieldTypeV2.java:7]: The previous fieldType used everything except `TEXT` and https://appdirect.jira.com/wiki/spaces/PI/pages/982647318/TD+Required+field+Service#3.4.1.-Read-vendor-fields has that extra fieldType

File: src/main/java/com/appdirect/sdk/vendorFields/model/VendorRequiredFieldsResponse.java [hunk 2]
```diff
@@ -13,6 +13,5 @@
 @AllArgsConstructor
 @JsonInclude(JsonInclude.Include.NON_EMPTY)
 public class VendorRequiredFieldsResponse {
-
```
Comment (abidhkhan) [inline src/main/java/com/appdirect/sdk/vendorFields/model/VendorRequiredFieldsResponse.java:16]: Remove this Line

**Comment by abidhkhan** [comment]
@AjayAujlaWork  please get this merged  by @naponce  and create release version

**Comment by abidhkhan** [comment]
@naponce   could you please merge and create a release version for this.

**Comment by naponce** [comment]
@abidhkhan Please, confirm if I can proceed with merge and release or other changes are expected.

**Comment by AjayAujlaWork** [comment]
> I don't see POST API changes?

Not sure what you mean. The PR has the following:

POST for /api/v2/admin/vendorValidations
https://github.com/AppDirect/service-integration-sdk/pull/217/files?diff=unified&w=1#diff-718e2f3aa1c28663b9ea65b7286a7920R70

GET for /api/v2/admin/requiredFields
https://github.com/AppDirect/service-integration-sdk/pull/217/files?diff=unified&w=1#diff-ee5db69f1d69bd59751c8bf07702c8b7R65

File: src/main/java/com/appdirect/sdk/vendorFields/controller/VendorFieldValidationController.java [hunk 3]
```diff
@@ -34,36 +36,76 @@
...
 
-	@RequestMapping(
-			method = POST,
-			value = "/api/v1/admin/vendorValidations",
-			consumes = APPLICATION_JSON_VALUE,
-			produces = APPLICATION_JSON_VALUE)
-	public Callable<VendorFieldsValidationResponse> validateFields(
-			@RequestBody VendorFieldsValidationRequest vendorFieldsValidationRequest,
-			@RequestHeader(value = "Accept-Language") List<Locale> locales) {
+    @Autowired
```
Comment (naponce) [inline src/main/java/com/appdirect/sdk/vendorFields/controller/VendorFieldValidationController.java:40]: No need to use `@Autowired` if `@RequiredArgsConstructor` is used in the class

File: src/main/java/com/appdirect/sdk/vendorFields/controller/VendorFieldValidationController.java [hunk 4]
```diff
@@ -34,36 +36,76 @@
...
-		log.info(
-				"Calling validate fields API with editionCode={}, flowType={}, operationType={}, partner={}, applicationIdentifier={}, locales={}",
-				vendorFieldsValidationRequest.getEditionCode(),
-				vendorFieldsValidationRequest.getFlowType(),
-				vendorFieldsValidationRequest.getOperationType(),
-				vendorFieldsValidationRequest.getPartner(),
-				vendorFieldsValidationRequest.getApplicationIdentifier(),
-				locales
-		);
+    @Autowired
```
Comment (naponce) [inline src/main/java/com/appdirect/sdk/vendorFields/controller/VendorFieldValidationController.java:43]: Same here, you can remove `@Autowired`

File: src/main/java/com/appdirect/sdk/vendorFields/controller/VendorFieldValidationController.java [hunk 5]
```diff
@@ -34,36 +36,76 @@
...
+                        "applicationIdentifier={}, " +
+                        "editionId={}, " +
+                        "flowType={}, " +
+                        "operationType={}, " +
+                        "userId={}, " +
+                        "companyId={}, " +
+                        "salesAgentUserId={}, " +
+                        "salesAgentCompanyId={}, " +
+                        "partnerCode={}, " +
+                        "is not supported.",
```
Comment (naponce) [inline src/main/java/com/appdirect/sdk/vendorFields/controller/VendorFieldValidationController.java:89]: Remove `"is not supported."` to this log message

File: src/main/java/com/appdirect/sdk/vendorFields/controller/VendorRequiredFieldController.java [hunk 6]
```diff
@@ -33,34 +34,90 @@
...
-	@RequestMapping(
-			method = GET,
-			value = "/api/v1/admin/requiredFields",
-			produces = APPLICATION_JSON_VALUE)
-	public Callable<VendorRequiredFieldsResponse> getRequiredFields(
-			@RequestParam(value = "editionCode") String editionCode,
-			@RequestParam(value = "flowType") FlowType flowType,
-			@RequestParam(value = "operationType") OperationType operationType,
-			@RequestHeader(value = "Accept-Language") List<Locale> locales) {
+    @Autowired
```
Comment (naponce) [inline src/main/java/com/appdirect/sdk/vendorFields/controller/VendorRequiredFieldController.java:38]: Remove `@Autowired`

File: src/main/java/com/appdirect/sdk/vendorFields/controller/VendorRequiredFieldController.java [hunk 7]
```diff
@@ -33,34 +34,90 @@
...
+    private final VendorRequiredFieldHandler vendorRequiredFieldHandler;
 
+    @Autowired
```
Comment (naponce) [inline src/main/java/com/appdirect/sdk/vendorFields/controller/VendorRequiredFieldController.java:41]: Remove `@Autowired`

**Comment by AjayAujlaWork** [comment]
> @abidhkhan Please, confirm if I can proceed with merge and release or other changes are expected.

https://appdirect.slack.com/archives/G012XMHFDME/p1591881193036200?thread_ts=1591867026.029200&cid=G012XMHFDME

@naponce @abidhkhan

**Pull request merged**